list integration test: error scanning 'restic list blobs'

wplapper opened the pull request:
func parseIDsFromReader does now scans the output from 'restic list blobs' correctly. 'list blobs' produces the blob type as well as the blob ID.

The test 'TestListBlobs' verifies the modified functionality. Since 'testListSnapshots' is used universally, there is no extra new test for the existing functionality.

<!--
Thank you very much for contributing code or documentation to restic! Please
fill out the following questions to make it easier for us to review your
changes.
-->

What does this PR change? What problem does it solve?
-----------------------------------------------------
This PR fixes a problem when using `restic list blobs` in the test environment. The output of `restic list blobs` contains currently 2 word, the blob type and the blob ID itself.
<!--
Describe the changes and their purpose here, as detailed as needed.
-->

Was the change previously discussed in an issue or on the forum?
----------------------------------------------------------------
No.
<!--
Link issues and relevant forum posts here.

If this PR resolves an issue on GitHub, use "Closes #1234" so that the issue
is closed automatically when this PR is merged.
-->

Checklist
---------

<!--
You do not need to check all the boxes below all at once. Feel free to take
your time and add more commits. If you're done and ready for review, please
check the last box. Enable a checkbox by replacing [ ] with [x].

Please always follow these steps:
- Read the [contribution guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches).
- Enable [maintainer edits](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- Run `gofmt` on the code in all commits.
- Format all commit messages in the same style as [the other commits in the repository](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits).
-->

- [X] I have added tests for all code changes.
- [ ] I have added documentation for relevant changes (in the manual).
- [ ] There's a new file in `changelog/unreleased/` that describes the changes for our users (see [template](https://github.com/restic/restic/blob/master/changelog/TEMPLATE)).
- [X] I'm done! This pull request is ready for review.

MichaelEischer commented on cmd/restic/cmd_list_integration_test.go:
```diff
@@ -42,3 +57,34 @@ func testListSnapshots(t testing.TB, opts GlobalOptions, expected int) restic.ID
+	resticIDs := testRunList(t, "blobs", env.gopts)
+	testIDSet := restic.NewIDSet(resticIDs...)
+
+	// get repo
+	_, repo, unlock, err := openWithReadLock(context.TODO(), env.gopts, false)
+	rtest.OK(t, err)
+	defer unlock()
+
+	setFromIndex := restic.IDSet{}
+	// make sure the index is loaded
+	rtest.OK(t, repo.LoadIndex(context.TODO(), nil))
+	// get first tree blob
+	rtest.OK(t, repo.ListBlobs(context.TODO(), func(blob restic.PackedBlob) {
+		setFromIndex.Insert(blob.ID)
+	}))
```
Please split this into a helper function like the following.
```
func testListBlobs(t testing.TB, gopts GlobalOptions) restic.IDSet
```

A function should stick to functions from the same abstraction level. The `testRunBackup` etc. functions are rather high-level, whereas `repo.LoadIndex` etc. is much more low level. Splitting these makes it easier to reuse code in the future. (And yes I'm aware that the tests are massively violating that rule of thumb, unfortunately).

wplapper commented:
done.

wplapper committed:
restic list blobs

Had to restart from scratch since the interface to all the commands had changed.
At the same time I upgraded to restic 0.18.1-dev

MichaelEischer committed:
fix repository usage in test

MichaelEischer reviewed (approved):
LGTM.

I've fixed the repository usage to still hold the lock while listing the blobs
